File: ButtonWithActionListener.py
import java
import sys
from java.awt import EventQueue
from java.awt.event import ActionListener
from java.lang import Runnable
from javax.swing import JButton
from javax.swing import JFrame


class ButtonWithActionListener(Runnable, ActionListener):
    def run(self):
        frame = JFrame('ButtonDemo_02')
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE)
        button = frame.add(JButton('Press me'))
        button.addActionListener(self)
        frame.pack()
        frame.setVisible(1)

    def actionPerformed(self, e):
        print 'button pressed'


if __name__ in ['__main__', 'main']:
    EventQueue.invokeLater(ButtonWithActionListener())
    if 'AdminConfig' in dir():
        raw_input('\nPress <Enter> to terminate the application:\n')
else:
    print '\nError: This script should be executed, not imported.\n'
    which = ['wsadmin -f', 'jython']['JYTHON_JAR' in dir(sys)]
    print 'Usage: %s %s.py' % (which, __name__)
    sys.exit()

File: Buttons/ButtonWithActionListener.java
import java.awt.event.*;
import javax.swing.*;

public class ButtonWithActionListener {
    public static void main( String[] args ) {
        javax.swing.SwingUtilities.invokeLater( new Runnable() {
            public void run() {
                JFrame frame = new JFrame( "ButtonDemo" );
                frame.setDefaultCloseOperation( JFrame.EXIT_ON_CLOSE );
                JButton button = new JButton( "Press me" );
                button.addActionListener( new ActionListener() {
                    public void actionPerformed( ActionEvent ae ) {
                        System.out.println( "button pressed" );
                    };
                } );
                frame.add( button );
                frame.pack();
                frame.setVisible( true );
            }
        });
    }
}